synapse/api/auth.py

```diff
@@ -208,6 +209,18 @@ async def get_user_by_req(
                     user_agent=user_agent,
                     device_id=device_id,
                 )
+                # Track also the puppeted user client IP if enabled and the user is puppeting
+                if (
+                    user_info.user_id != user_info.token_owner
+                    and self._mau_track_puppeted_users
+                ):
+                    await self.store.insert_client_ip(
+                        user_id=user_info.user_id,
+                        access_token=access_token,
+                        ip=ip_addr,
+                        user_agent=user_agent,
+                        device_id=device_id,
+                    )
```

Comment:
`insert_client_ip` also affects the list of client ips and devices, which aren't directly related to MAU. I think we should either call `populate_monthly_active_users` directly, or rename the config option to be clearer about the side effects.

edit: `populate_monthly_active_users` isn't available on workers unfortunately

Reply:
Yeah I was struggling a bit on how to submit this due to exactly this reason that feels a bit wrong. As a change it feels like the least invasive change to how things currently work and behave though, so I went with this suggestion.

There are other settings that affect MAU that are outside the MAU config settings, like `track_appservice_user_ips`. We could do the same here. Possibly `ServerConfig` item (on the top level) as `track_puppeted_user_ips` (logical because it also contains MAU config items). Would that sound better or do you have other suggestions where the config would logically live?

Reply:
I wasn't aware we had `track_appservice_user_ips`, thanks! I'd be in favour of us following the same pattern and going with `track_puppeted_user_ips`.

Reply:
Have adjusted! Moved the config flag to api config, which seems more relevant, given this is used in api auth. The server config feels a bit bloated already with different things and the appservice option was in the appservice config, so these couldn't be bundled.

Should years in copyright be updated btw when changing files?

Reply:
> Should years in copyright be updated btw when changing files?

I think so, but I'm not going to block the PR over it!